debugging-modules/README.md

```diff
@@ -0,0 +1,264 @@
+> with [WebAssembly Interface
+> Types](https://github.com/WebAssembly/interface-types/). However, since that
+> standard is still coming together, we are temporarily describing the
+> interfaces with Web IDL.
+
+The definitions for interfaces defined in this proposal are available in
+[`debugging-modules.webidl`](./debugging-modules.webidl).
+
+## FAQ
+
+### Why not an existing debug info format like DWARF or PDB?
+
+There are a variety of reasons:
+
+* Future extensibility is harder with data formats than interfaces. With
+  interfaces, we can always add new, optional methods that debugging modules can
+  implement and use or not. With data formats, we need to design future
+  extensibility into the format, which requires more care and can also bloat its
+  encoded size with repetitive metadata and field tags.
+
+* Interpreting DWARF and PDB data is not straight forward, and the full
+  responsibility for that would lie completely on developer tools authors. In
+  contrast, debugging modules provide high-level methods for querying program
+  information, which are easier to work with for developer tools
+  authors. Toolchains can wrap DWARF or PDB interpreters in a debugging module
+  interface, and this debugging module implementation can be shared across any
+  toolchains that use that same format. Sharing debugging module implementations
+  is easier for toolchains than sharing DWARF/PDB interpretation logic is for
+  developer tools, since debugging modules talk to a `WasmDebugger` interface
+  and produce a well-known `SourceDebugger` interface that are used by all
+  toolchains, while developer tools would be calling into internal APIs that
+  aren't common across developer tools implementations.
+
+* Both debug info formats and standards ossify and evolve slowly. With debugging
+  modules, we allow interface implementations to experiment and iterate
+  separately from the standard. This allows innovations (for example more
+  compact data representations) to develop and ship without needing to go
+  through the slow standards process.
+
+* Static debug info formats don't provide a path forward for source-level
+  debugging of projects like Blazor, that run interpreted code inside their
+  Wasm. Debugging that interpreted code requires dynamic debugging APIs.
+
+### Why not a protocol?
+
+A wire protocol requires defining the same set of operations we want to support
+that we define as interface methods in this proposal, and *also* a serialization
+format. Defining a serialization format that is both compact and
+future-extensible is no small task. Additionally, nothing about source-level
+debugging *requires* over-the-wire communication or message passing, even if
+that is often a good architectural decision. Implementations are free to proxy
+this proposal's interface method calls across a protocol or to another
+process. It doesn't make sense to bake a specific wire protocol into the
```

Comment:
I think that implementations will always need to proxy the interface method calls across a protocol, because the debugger and debuggee will always be in different processes.
This does not mean that we necessarily need to add a specific wire protocol into the standard, but the standard should mention how the Debugger Module will communicate with the debuggee.